src/lib/components/01-elements/button/button.tsx

```diff
@@ -68,6 +68,7 @@ export default function Button({
   variant = 'solid',
   children,
   className,
+  as,
```

Comment:
We need to extract the `as` prop from the `props` variable, because the `props` variable is being spread in `<Element className={defaultClasses} {...props}>`.
Because `Element` could be a `next-link`, we were overloading a the `as` prop in the `Link` element. The `Link.as` prop do the following:

> The `as` prop in the `Link` component allows you to specify the URL path for the browser's address bar while still rendering a different page component. This is useful in scenarios where you want to have a pretty URL displayed in the browser's address bar while loading content from a different page.

However, this prop has issues in [nextjs app router](https://github.com/vercel/next.js/issues/44146#issuecomment-1357742761). So, what was happening is that links with the  `Link` element, no matter which `href` would be passed, the link would always be `/link` because in [here](https://github.com/fourkitchens/forcepoint-shared-components/pull/33/files#diff-e5d2bd726805ee5f2e5cd5220561f4ff9c07426f5ebe1d5c18057924e71066d2R131) we're saying the element should be of type link.